editions/tw5.com/tiddlers/widgets/DataWidget.tid

```diff
@@ -0,0 +1,65 @@
+caption: data
+created: 20230406161341763
+modified: 20230406161341763
+tags: Widgets
+title: DataWidget
+type: text/vnd.tiddlywiki
+
+! Introduction
+
+The data widget is used with the <<.wlink TestCaseWidget>> widget and the [[Innerwiki Plugin]] to specify payload tiddlers that are to be included in the test case or innerwiki.
+
+! Content and Attributes
+
+The content of the data widget is rendered as if the data widget were not present. It supports the following attributes:
+
+|!Attribute |!Description |
+|<<.attr $tiddler>> |Optional title of a tiddler to be used as a payload tiddler (optional) |
+|<<.attr $filter>> |Optional filter string identifying tiddlers to be used as payload tiddlers (optional) |
+|<<.attr $compound-tiddler>> |Optional title of a tiddler containing payload tiddlers in `text/vnd.tiddlywiki-multiple` format (see below) |
+|//any attribute<br>not starting<br>with $// |Field values to be assigned to the payload tiddler(s) |
+
+The data widget is not rendered when used within the <<.wlink TestCaseWidget>> widget or the [[Innerwiki Plugin]] but for ease of testing, when used elsewhere it renders a JSON representation of the payload tiddlers.
```

Comment:
> when used elsewhere it renders a JSON representation of the payload tiddlers

I'm not able to get this behavior to work. Outside of the testcase widget, I find the data widget to render empty.

Reply:
Thanks @btheado I had replaced that feature of the data widget with the ability for the testcase widget to display its payload in JSON via a special template. I've pushed an update with the docs in 0db4c449397c0b30498c5e052e9fcb80bed90541 that explains how to do that. I've also added the JSON output feature back to the data widget in 1d89c7925aa8392b450e7ef36c04fc3f360f9173 with tests in 0db4c449397c0b30498c5e052e9fcb80bed90541